sys/arm64/include/param.h

```diff
@@ -47,8 +47,17 @@
 #ifndef MACHINE
 #define	MACHINE		"arm64"
 #endif
-#ifndef MACHINE_ARCH
-#define	MACHINE_ARCH	"aarch64"
+#if !__has_feature(capabilities)
+# ifndef MACHINE_ARCH
+#  define	MACHINE_ARCH	"aarch64"
+# endif
+#else
+# ifndef MACHINE_ARCH
+#  define	MACHINE_ARCH	"morello"
```

Comment:
I think we want this to be `aarch64c`, otherwise it's going to be a weird special case.

Reply:
We decided to not use aarch64c so it could be used if a production aarch64-based ISA was ever developed.

Reply:
But then it differs from what the Makefile MACHINE_ARCH/TARGET_ARCH is.

Reply:
MACHINE_ARCH will be morello

Reply:
MACHINE_ARCH will be either `aarch64` for legacy/hybrid or `morello` for CheriABI on Morello

Reply:
I know there are places we assume things like the ABI being 64-bit if and only if MACHINE_ARCH matches `*64*`. If you want something special, at least have `aarch64` in there.

Reply:
Then the checks are broken, see also why most browsers claim to be Mozilla.

Reply:
Sure, but the less special-casing we need to do for morello because it wants to be different the better.

Reply:
There are 2 places where we check for `*64*`. The `libarchive` check is from upstream and could probably be a check for `MACHINE_ABI`. The `bsd.cpu.mk` change sets `MACHINE_ABI`, so would be the only special case we need.

```
% ag "(MACHINE|TARGET)_ARCH.*\*64\*"
lib/libarchive/Makefile
40:.if ${MACHINE_ARCH:M*64*} == ""

share/mk/bsd.cpu.mk
457:.if ${MACHINE_ARCH:M*64*}
```